p4-16/psa/examples/psa-example-resubmit2.p4

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2017 Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <core.p4>
+#include <psa.p4>
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+header ipv4_t {
+    bit<4>  version;
+    bit<4>  ihl;
+    bit<8>  diffserv;
+    bit<16> totalLen;
+    bit<16> identification;
+    bit<3>  flags;
+    bit<13> fragOffset;
+    bit<8>  ttl;
+    bit<8>  protocol;
+    bit<16> hdrChecksum;
+    bit<32> srcAddr;
+    bit<32> dstAddr;
+}
+
+header header_a_t {
+    bit<32> field_a;
+}
+
+header header_b_t {
+    bit<16> field_b;
+    bit<16> field_c;
+}
+
+header_union resubmit_h {
+    header_a_t h_a;
+    header_b_t h_b;
+}
+
+// BEGIN:Resubmit_Example_Part1
+struct resubmit_metadata_t {
+    bit<8> selector;
+    resubmit_h h_u;
+}
+// END:Resubmit_Example_Part1
+
+struct empty_metadata_t {}
+
+struct fwd_metadata_t {
+    bit<9> output_port;
+}
+
+struct metadata {
+    header_a_t h_a;
+    header_b_t h_b;
+    bit<8> resubmit_id;
+    fwd_metadata_t fwd_meta;
+}
+
+struct headers {
+    ethernet_t       ethernet;
+    ipv4_t           ipv4;
+}
+
+parser CommonParser(packet_in buffer,
+    out headers parsed_hdr,
+    inout metadata user_meta)
+{
+    state start {
+        transition parse_ethernet;
+    }
+    state parse_ethernet {
+        buffer.extract(parsed_hdr.ethernet);
+        transition select(parsed_hdr.ethernet.etherType) {
+            0x0800: parse_ipv4;
+            default: accept;
+        }
+    }
+    state parse_ipv4 {
+        buffer.extract(parsed_hdr.ipv4);
+        transition accept;
+    }
+}
+
+parser IngressParserImpl(
+    packet_in buffer,
+    out headers parsed_hdr,
+    inout metadata user_meta,
+    in psa_ingress_parser_input_metadata_t istd,
+    in resubmit_metadata_t resub_meta,
+    in empty_metadata_t recirculate_meta)
+{
+    CommonParser() cp;
+
+    state start {
+        transition select(istd.packet_path) {
+           PSA_PacketPath_t.RESUBMIT: copy_resubmit_meta;
+           PSA_PacketPath_t.NORMAL: packet_in_parsing;
+        }
+    }
+
+    state copy_resubmit_meta {
+        transition select(resub_meta.selector) {
+           8w0 : parse_resubmit_0;
+           8w1 : parse_resubmit_1;
+        }
```

Comment:
So if you are planning to use a header_union in an example, there shouldn't really be any need to have separate assignments for the cases where one member of the header_union is valid vs. a different member.  You can simply have a single assignment like `user_meta.h_u = resub_meta.h_u;`, where `user_meta.h_u` has the same header_union type as `resub_meta.h_u`.

I am not saying anything is functionally wrong with this code, but why write more code when something shorter will achieve the same effect?